src/ACEtk/block/EquiprobableOutgoingEnergyBinData/src/generateXSS.hpp

```diff
@@ -0,0 +1,35 @@
+static std::vector< double >
+generateXSS(
+  std::vector< long >&& boundaries,
+  std::vector< long >&& interpolants,
+  std::vector< EquiprobableOutgoingEnergyBins >&& distributions ) {
+
+  // reserve size in the vector
+  std::size_t nr = boundaries.size();
+  std::size_t ne = distributions.size();
+  std::vector< double > xss( 2 + ne ); // NE, list of energies, NET
```

Comment:
[comment]
This way of constructing XSS appears to be 100% correct. However, I had to spend more time than I'd like proving it to myself due to the nonlinearity going on (allocate, prepend, set values, append).